app/views/layouts/listnav/_host.html.haml

```diff
@@ -28,78 +28,46 @@
-          %li.disabled
-            = link_to(_("Storage Adapters"), "#")
-        - else
-          %li
-            = link_to(_('Storage Adapters'),
-              {:action  => 'show', :id => @record, :display => 'storage_adapters'},
-              :title => _("Show %s storage adapters") % host_title)
-
-        - if @osinfo.nil? || @osinfo.empty?
-          %li.disabled
-            = link_to(_("OS Information"), "#")
-        - else
-          %li
-            = link_to(_('OS Information'),
-              {:action  => 'show', :id => @record, :display => 'os_info'},
-              :title => _("Show %s OS information") % host_title)
-
-        - if @vmminfo.nil? || @vmminfo.empty?
-          %li.disabled
-            = link_to(_('VMM Information'), "#")
-        - else
-          %li
-            = link_to(_('VMM Information'),
-              {:action => 'show', :id => @record, :display => 'hv_info'},
-              :title => _("Show VMM information"))
-
-        - if @record.has_perf_data?
-          %li
-            = link_to(_('Capacity & Utilization'),
-              {:action => 'show', :id => @record, :display => 'performance'},
-              :title => _("Show Capacity & Utilization"))
-        - else
-          %li.disabled
-            = link_to(_('Capacity & Utilization'), "#")
-
-        - if @record.has_events? || @record.has_events?(:policy_events)
-          %li
-            = link_to('Timelines',
-              {:action => 'show', :display => 'timeline', :id => @record},
-              :title => _("Show Timelines"))
-        - else
-          %li.disabled
-            = link_to(_('Timelines'), "#")
-
+        = li_link_if(:if        => !(@devices.nil? || @devices.empty?),
+                     :text      => _('Devices'),
+                     :record_id => @record.id,
+                     :display   => 'devices',
+                     :title     => _("Show %s devices") % host_title)
+        = li_link_if_nonzero(:count     => @record.number_of(:switches),
+                             :text      => _('Network'),
+                             :record_id => @record.id,
+                             :display   => 'network',
+                             :title     => _("Show %s network") % host_title)
+        = li_link_if(:if        => !(@record.hardware.blank? || @record.hardware.number_of(:storage_adapters) == 0),
+                     :text      => _('Storage Adapters'),
+                     :record_id => @record.id,
+                     :display   => 'storage_adapters',
+                     :title     => _("Show %s storage adapters") % host_title)
+        = li_link_if(:if        => !(@osinfo.nil? || @osinfo.empty?),
+                     :text      => _('OS Information'),
+                     :record_id => @record.id,
+                     :display => 'os_info',
+                     :title => _("Show %s OS information") % host_title)
```

Comment:
here and above, please use the named key for the format string to help the translators